### ISSUE TITLE
InjectedAuth not working properly

### ISSUE THREAD

**Issue by nixcode1** [GIfatahTH/states_rebuilder#201]
Hello, I just discovered this package and I'm loving it. I did run into some problems though.
Not sure if its a bug or I'm just not using it right but when unsignedUser is not null, restarting the app always goes to the homepage even after signing out.

```dart
//When unsignedUser is commented out, auth works as it should and the right page is displayed depending on the last action taken(Sign in or Sign out)
final InjectedAuth<User?, AuthParam> user = RM.injectAuth<User?, AuthParam>(
    () => UserRepository(),
    // unsignedUser: UnsignedUser(),
    onSigned: (_) => print("Signed"),
    onUnsigned: () => print("Unsigned"),
    onAuthStream: (repo) => (repo as UserRepository).currentUser().asStream(),
    onSetState: On.error(
      (err, refresh) {
        if (err is EmailException || err is PasswordException) {
          return;
        }
        RM.navigate.toDialog(
          AlertDialog(
            title: Text(ExceptionsHandler.errorMessage(err).title!),
            content: Text(ExceptionsHandler.errorMessage(err).message!),
          ),
        );
      },
    ),
    debugPrintWhenNotifiedPreMessage: "Auth");```


//user is always signed it even when signed out.

final InjectedAuth<User?, AuthParam> user = RM.injectAuth<User?, AuthParam>(
    () => UserRepository(),
    unsignedUser: UnsignedUser(),
    onSigned: (_) => print("Signed"),
    onUnsigned: () => print("Unsigned"),
    onAuthStream: (repo) => (repo as UserRepository).currentUser().asStream(),
    onSetState: On.error(
      (err, refresh) {
        if (err is EmailException || err is PasswordException) {
          return;
        }
        RM.navigate.toDialog(
          AlertDialog(
            title: Text(ExceptionsHandler.errorMessage(err).title!),
            content: Text(ExceptionsHandler.errorMessage(err).message!),
          ),
        );
      },
    ),
    debugPrintWhenNotifiedPreMessage: "Auth");
```

The code is a copy of the firebase example

**Comment by GIfatahTH** [comment]
Hi, @nixcode1 welcome to states_rebuilder, and thank you for opening this issue.

In the example, the User is nullable, `InjectedAuth<User?, AuthParam>`, this means that null is considered as the unsigned user.

To explicitly set the unsignedUser, the User must be non-nullable, `InjectedAuth<User, AuthParam>`.

The first case, (unsignedUser is commented), works because the onAuthStream emits a null which is interpreted as unsigned user.

Whereas, in the second case, the  onAuthStream return null, but as unsignedUser is not null, the null is interpreted as a signed user.

The bottom line, if you want to explicity defined the unsignedUser use non nullable User.

**Comment by nixcode1** [comment]
Thank you so much for the reply and for taking your time to properly explain it. I completely understand now. 

Please do keep up the good work you're doing with this package

**Comment by GIfatahTH** [comment]
@nixcode1 Happy with your issue,

I added an assertion error when setting non-null unsignedUser for nullable user.

Thanks again and looking forward to other issues.
